tests/client/test_query.py

```diff
@@ -8,181 +8,31 @@
-    etcd_internal_source = SourceLocation(url=etcd_internal_url).save()
-    etcd_build.source_location.connect(etcd_internal_source)
-    etcd_internal_source.upstream.connect(etcd_upstream_source)
-    etcd_component = Component(canonical_type='github', canonical_namespace='coreos',
-                               canonical_name='etcd').save()
-    etcd_upstream_source.component.connect(etcd_component)
-
-    # The container itself
-    container_build = Build(id_='4000', type_='container').save()
-    container_filename = 'docker-image-sha256:' + '1' * 64 + '.x86_64.tar.gz'
-    image = Artifact(
-        archive_id='4001',
-        architecture='x86_64',
-        filename=container_filename,
-        type_='container').save()
-    container_build.artifacts.connect(image)
-    container_internal_url = 'git://pkgs.domain.local/containers/image#...'
-    container_internal_source = SourceLocation(url=container_internal_url).save()
-    container_build.source_location.connect(container_internal_source)
-    image.embedded_artifacts.connect(parent)  # inherits from parent image
-    image.embedded_artifacts.connect(etcd_rpm)  # etcd RPM is installed
-
-    # Go executable
-    # Based on output from backvendor:
-    # *github.com/jimmidyson/configmap-reload@043045da[...] =v0.2.2 ~v0.2.2
-    # golang.org/x/sys@9c60d1c[...] ~v0.0.0-0.20150901164945-9c60d1c508f5
-    # gopkg.in/fsnotify.v1@c282820[...] =v1.4.7 ~v1.4.7
-    configmap_reload = SourceLocation(url='https://github.com/jimmidyson/configmap-reload',
-                                      canonical_version='v0.2.2').save()
-    configmap_reload.component.connect(
-        Component(canonical_type='golang',
-                  canonical_namespace='github.com/jimmidyson',
-                  canonical_name='configmap-reload').save())
-    x_sys = SourceLocation(url='https://go.googlesource.com/sys',
-                           canonical_version='v0.0.0-0.20150901164945-9c60d1c508f5').save()
-    x_sys.component.connect(
-        Component(canonical_type='golang',
-                  canonical_namespace='golang.org/x',
-                  canonical_name='sys').save())
-    fsnotify = SourceLocation(url='https://gopkg.in/fsnotify.v1',
-                              canonical_version='v1.4.7').save()
-    fsnotify.component.connect(
-        Component(canonical_type='golang',
-                  canonical_namespace='gopkg.in',
-                  canonical_name='fsnotify.v1').save())
-
-    container_internal_source.upstream.connect(configmap_reload)
-    container_internal_source.embedded_source_locations.connect(x_sys)
-    container_internal_source.embedded_source_locations.connect(fsnotify)
+    parent_build_id, build_id = UseCaseFactory.container_with_go_and_rpm_artifacts()
 
-    assert query.get_container_by_component('fsnotify.v1', 'golang', 'v1.4.7') == {4000}
+    assert query.get_container_by_component('fsnotify', 'golang', 'v1.4.7') == {build_id}
```

Comment:
Could you explain why '.v1' was removed? I assume it was an issue in previous test :-)

Reply:
Yes, should be "fsnotify.v1".

Reply:
No reason really, the factories use `fsnotify` so I chose to adjust the test instead. I can certainly switch it back to `v1` but it's just a name so I didn't think it mattered what it actually was.

Reply:
Changed!